stan/math/prim/mat/prob/bernoulli_logit_glm_lpmf.hpp

```diff
@@ -60,13 +61,13 @@ return_type_t<T_x, T_alpha, T_beta> bernoulli_logit_glm_lpmf(
   const size_t N = x.rows();
   const size_t M = x.cols();
 
-  check_bounded(function, "Vector of dependent variables", y, 0, 1);
   check_consistent_size(function, "Vector of dependent variables", y, N);
   check_consistent_size(function, "Weight vector", beta, M);
   if (is_vector<T_alpha>::value) {
     check_consistent_sizes(function, "Vector of intercepts", alpha,
                            "Vector of dependent variables", y);
   }
+  check_bounded(function, "Vector of dependent variables", y, 0, 1);
```

Comment:
What is the reason behind this move? So we dont check the vector if the sizes dont match?

Reply:
I think unmatching sizes is larger error than wrong values, so it should be checked and reported first. If you disagree I can just as well revert that.

Reply:
I agree with your reasoning, thanks.